src/handle_action.cpp

```diff
@@ -783,7 +783,10 @@ static void smash()
             int glove_coverage = 0;
             for( const item &i : u.worn ) {
                 if( ( i.covers( bodypart_id( "hand_l" ) ) || i.covers( bodypart_id( "hand_r" ) ) ) ) {
-                    int temp_coverage = i.get_coverage();
+                    int temp_coverage = i.get_coverage( bodypart_id( "hand_l" ) );
+                    if( temp_coverage == 0 ) {
+                        temp_coverage = i.get_coverage( bodypart_id( "hand_r" ) );
+                    }
```

Comment:
I thought this code was used to calculate damage based on smashing with limbs? Surely it should not calculate average and should instead look for the highest? Since you will only be smashing something with your gloved hand for example.